### DISCUSSION TITLE
Question RE: best way to organize multiple entrypoints to a common WDL code base

### DISCUSSION THREAD

**Discussion by NatPRoach** [aws/amazon-genomics-cli#487] [General]
Hello, I had what is hopefully a quick question about the best way to handle large projects with multiple WDL entry points in AGC.

If we have a bunch of workflows that are importing from common WDL task files etc, is there a best practices for defining those different workflows to AGC?

My initial approach was to have a staging directory where each workflow entrypoint `WDL` would be symlinked in a directory corresponding to that workflow and in that directory have `inputs.json` and `MANIFEST.yaml` files. The problem there being that it seems like AGC doesn't then follow those symlinks when doing imports for the WDL. (i.e. if my workflow WDL `MyExampleWorkflow` imports a task WDL file with `import "example_tasks.wdl"`, AGC will fail due to not being able to find `example_tasks.wdl` in the staging directory.)

Is there a preferred way of managing this that I'm not thinking of?

Thanks in advance!

**Comment by markjschreiber** [comment]
Following SIM links might be a feature request although there are some implications as AGC will traverse nested directories.

One possible work around would be to have a github repo of common WDL subworkflows and for each workflow you `git clone` that into a sub-directory of the workflow.
